pkg/kubernetes/service/v1alpha1/kubernetes.go

```diff
@@ -69,6 +84,13 @@ func (k *Kubeclient) withDefaults() {
 			return kubernetes.NewForConfig(config)
 		}
 	}
+	if k.getClientsetForPath == nil {
```

Comment:
there is refactoring happened for other k8s.go from @AmitKumarDas .. may be worth for you to look at that https://github.com/openebs/maya/pull/1318

Reply:
Done